### PR TITLE
🎉Source Gitlab: preserve data in records with flattened keys

### LINKED ISSUE #22124: GitLab Source: Excluding Data from Responses
## Environment
- **Airbyte version**:  0.40.32
- **OS Version / Instance**: N/A
- **Deployment**: N/A
- **Source Connector and version**: GitLab
- **Destination Connector and version**: N/A
- **Step where error happened**: Sync job 

## Current Behavior
The GitLab Source is eliminating certain nested objects when calling [ `flatten_id`](https://github.com/airbytehq/airbyte/blob/v0.40.32/airbyte-integrations/connectors/source-gitlab/source_gitlab/streams.py#L103). For example, issues have a `milestone` property that contains info about a linked milestone, but since milestone is listed in the `flatten_id_keys` prop for the `Issues` class, it all gets blown away in favor of adding a `milestone_id` property to the main payload. This makes it pretty difficult to determine if the milestone is a group or project milestone, which is easily identifiable when the full `milestone` property is on the message.

So currently the record for an issue looks like:

```json
{
    "type": "RECORD",
    "record": {
        "stream": "issues",
        "data": {
            "id": 1,
            "iid": 1,
            "project_id": 1,
            "title": "foo",
            "description": "testing",
            "state": "opened",
            "created_at": "2023-01-30T21:25:53.094Z",
            "updated_at": "2023-01-30T21:25:53.094Z",
            "closed_at": null,
            "labels": [],
            "assignees": [],
            "type": "ISSUE",
            "user_notes_count": 0,
            "merge_requests_count": 0,
            "upvotes": 0,
            "downvotes": 0,
            "due_date": null,
            "confidential": false,
            "discussion_locked": null,
            "issue_type": "issue",
            "web_url": "https://gitlab.com/foo/bar/-/issues/1",
            "time_stats": {
                "time_estimate": 0,
                "total_time_spent": 0,
                "human_time_estimate": null,
                "human_total_time_spent": null
            },
            "task_completion_status": {
                "count": 0,
                "completed_count": 0
            },
            "weight": null,
            "blocking_issues_count": 0,
            "has_tasks": false,
            "_links": {
                "self": "https://gitlab.com/api/v4/projects/1/issues/1",
                "notes": "https://gitlab.com/api/v4/projects/1/issues/1/notes",
                "award_emoji": "https://gitlab.com/api/v4/projects/1/issues/1/award_emoji",
                "project": "https://gitlab.com/api/v4/projects/1",
                "closed_as_duplicate_of": null
            },
            "references": {
                "short": "#1",
                "relative": "#1",
                "full": "foo/bar/#1"
            },
            "severity": "UNKNOWN",
            "moved_to_id": null,
            "service_desk_reply_to": null,
            "epic_iid": null,
            "epic": null,
            "iteration": null,
            "author_id": 1,
            "assignee_id": null,
            "closed_by_id": null,
            "milestone_id": 1
        },
        "emitted_at": 1675128509279
    }
}
```

## Expected Behavior
Not sure if this is behaving as designed, but from a consumer perspective I feel it's unexpected to remove data from the output.  Here's the missing `milestone` property from the above example:

```json
{
  "milestone": {
      "id": 1,
      "iid": 1,
      "group_id": 1,
      "title": "bar",
      "description": "",
      "state": "active",
      "created_at": "2023-01-27T21:00:07.855Z",
      "updated_at": "2023-01-27T21:00:07.855Z",
      "due_date": null,
      "start_date": null,
      "expired": false,
      "web_url": "https://gitlab.com/groups/foo/-/milestones/1"
    }
}
```
This is a group milestone since `group_id` is present. If instead it were `project_id` we would know it's a project milestone. These milestone types have different api endpoints so simply knowing the `milestone_id` isn't enough to look up the data.

## Are you willing to submit a PR?

Yup. Happy to submit a PR if this isn't a design decision.

### PR DESCRIPTION
**Pull request by BrentSouza**
## What

Writing this to resolve #22124

## How

Certain ids for nested objects are being "flattened" into the parent object, but the function is using `pop` to access the nested object, which eliminates useful information from the record. Instead of using `pop`, I'm proposing to just use `get` which so we can keep the "flattened" ids in the parent object for backward compatibility while preserving the original structure of the record.

## Recommended reading order

This is a simple 3 char swap in a single file. I recommend you read left to right 😉 

## 🚨 User Impact 🚨
This should not break anything for existing consumers. The flattened_ids are still maintained, we're just adding data or perhaps keeping data is the better term.

## Pre-merge Checklist

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by juweins — COMMENTED** [review]
From my perspective:

You already mentioned it in your issue statement. From a consumers point of view, I would expect to receive data as requested from the source. So changing to .get makes sense to me.

I share your thoughts on this:
> Was .pop a design decision?

However, since .get returns a None in some cases: Did you test this case? 
With only limited experience with Gitlab: How likely is that case? I see that there is no exception handling on _flatten_id so I would assume that there is no such case.

Thoughts please :)

**Comment by juweins** [comment]
Thanks for pointing this out to me! 👍🏼

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-gitlab/source_gitlab/schemas/issues.json [hunk 1]
```diff
@@ -98,15 +98,31 @@
...
+    "assignee": {
+      "type": ["null", "object"],
+      "additionalProperties": true
+    },
     "assignee_id": {
       "type": ["null", "integer"]
     },
+    "closed_by": {
+      "type": ["null", "object"],
+      "additionalProperties": true
```
Comment (archangelic) [inline airbyte-integrations/connectors/source-gitlab/source_gitlab/schemas/issues.json:117]: I was going to add all the properties into these, but many of them are only documented in examples outputs, or not at all.

**Comment by marcosmarxm** [comment]
The Oauth token got expired for today. Waiting instructions how to get this generated to make tests work again.

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**